src/cmds/qmgr.c

```diff
@@ -2005,6 +2005,11 @@ display(int otype, int ptype, char *oname, struct batch_status *status,
 				printf("Server %s\n", status->name);
 		}
 		else if (otype == MGR_OBJ_SCHED) {
+			if(oname && *oname && strncmp(oname,status->name,PBS_MAXSCHEDNAME)) {
```

Comment:
oname is passed into the function so please do the NULL checks on it at the top of the function itself, it'll make it more robust.

Reply:
Also, please add a space after commas inside the strncmp() call

Reply:
I agree NULL check for parameters should be done at the beginning of a function to make it robust.
However, it wont suit for this scenario. Due to following interrelated reasons:
   

1.  There a places of display() invocation with "oname" argument set to NULL, ex. in execute() of qmgr.c
2. I cannot fathom what to put under if (oname ==NULL) {} that is to be added at the begining.

Reply:
Please see my comment to below about 'oname', I don't think you need this if statement at all.

Reply:
I disagree.  It is definitely required, as explained in the logic below, we will be getting all the sched objects, so if the user gives "list sched sc_abc",  the `status` will be linked list of "default_sched"-->"sc_abc"-->"sc__xyz"-->"etc".  So now in this case, except for "sc_abc" we need to advance the status to next.

However if the user gives "list sched",  we have execute the while loop for all the sched.

Reply:
Ok, so the server (pbs_statsched call) will not do any filtering, it is expected that it will happen at the client side, got it, thanks for clarifying.